scripts/mixins/families/marid.lua

```diff
@@ -0,0 +1,29 @@
+require('scripts/globals/mixins')
+
+g_mixins = g_mixins or {}
+g_mixins.families = g_mixins.families or {}
+
+g_mixins.families.marid = function(maridMob)
+    -- 20% chance to break tusk on critical hit
+    maridMob:addListener('CRITICAL_TAKE', 'MARID_CRITICAL_TAKE', function(mob)
+        if not utils.chance(20) then
+            return
+        end
+
+        local broken = mob:getLocalVar('TuskBroken')
+        if broken < 2 then
+            broken = broken + 1
+            mob:setLocalVar('TuskBroken', broken)
+            mob:setAnimationSub(broken)
+        end
+    end)
+
+    maridMob:addListener('ITEM_DROPS', 'MARID_ITEM_DROPS', function(mob, loot)
+        local broken = mob:getLocalVar('TuskBroken')
+        for _ = 1, broken do
+            loot:addItem(xi.item.MARID_TUSK, xi.drop_rate.GUARANTEED)
+        end
+    end)
+end
```

Comment:
If animationSub happens to be the same as what the local var will be, couldnt we just check for the animationSub?

Reply:
Like so...

```lua
g_mixins.families.marid = function(maridMob)
    -- 20% chance to break tusk on critical hit
    maridMob:addListener('CRITICAL_TAKE', 'MARID_CRITICAL_TAKE', function(mob)
        local brokenTusks = mob:getAnimationSub()

        if
            brokenTusks < 2 and
            math.random(1, 100) <= 20
        then
            brokenTusks = brokenTusks + 1
            mob:setAnimationSub(brokenTusks)
        end
    end)

    maridMob:addListener('ITEM_DROPS', 'MARID_ITEM_DROPS', function(mob, loot)
        local brokenTusks = mob:getAnimationSub()

        for _ = 1, brokenTusks do
            loot:addItem(xi.item.MARID_TUSK, xi.drop_rate.GUARANTEED)
        end
    end)
end
```